lib/services/role.go

```diff
@@ -713,8 +713,8 @@ type AccessChecker interface {
 	// RoleNames returns a list of role names
 	RoleNames() []string
 
-	// CheckAccessToServer checks access to server.
-	CheckAccessToServer(login string, server types.Server, mfa AccessMFAParams) error
+	// CheckAccess checks access to the specified resource.
+	CheckAccess(r AccessCheckable, mfa AccessMFAParams, matchLabels bool, matchers ...RoleMatcher) error
```

Comment:
Would be nice to replace the `bool` with an `enum` or maybe split this into `CheckAccessWithLabels` and `CheckAccessWithoutLabels` that way the intent is a bit more clear to callers. Although this interface alone has several other functions with boolean parameters.

Reply:
I agree. It also feels weird that matchLabels is separate from other matches for some reason. Can it just be one of the matchers?

Reply:
Will make an enum for this. We might be able to remove it altogether with your confirmation on https://github.com/gravitational/teleport/pull/8407#discussion_r718799605